pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_sni.go

```diff
@@ -234,6 +235,26 @@ func (s *sni) Deploy(ctx context.Context) error {
 		return err
 	}
 
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, s.client, accessControlVPNServer, func() error {
+		accessControlVPNServer.Labels = getLabels()
+		accessControlVPNServer.Spec = istioapisecurityv1beta1.AuthorizationPolicy{
+			Action: accessControlSpec.Action,
+			Rules:  accessControlSpec.Rules,
+			Selector: &istiov1beta1.WorkloadSelector{
+				MatchLabels: s.values.IstioIngressGateway.Labels,
+			},
+		}
+		for i := range accessControlVPNServer.Spec.Rules {
+			accessControlVPNServer.Spec.Rules[i].When = []*istioapisecurityv1beta1.Condition{{
+				Key:    "request.headers[reversed-vpn]",
+				Values: []string{fmt.Sprintf("outbound|1194||vpn-seed-server.%s.svc.cluster.local", s.namespace)},
```

Comment:
That's really good you found that out! I am wondering though... can we rely on all of this being static values? Could become hard to maintain in the future.

Reply:
i checked everything on s.values but sadly i couldnt figure out how to make this more dynamic... i think we should go with this and see what the gardener guys say :)

Reply:
Alright then.